### PR TITLE
feat(embodied): add fullshard and full-param tune for pi

### PR DESCRIPTION
**Pull request by LiuZhihao2022**
### Description
(1) Adapt the OpenPI model to support full-parameter fine-tuning under the FSDP framework, with compatibility for full-shard mode.
(2) Support gradient norm clipping with mixed-precision gradients. Mainly follows the official PyTorch FSDP implementation (FullyShardedDataParallel source).

### Additional information (optional, e.g., figures and logs):

Experimental evaluation on LIBERO-Goal using π₀ shows that both the training curves and training metrics are consistent with the original results.

6a8d48b1-1081-416c-89b8-c3172dad6bdd

The original no_shard mode also runs correctly.
c6002688-0698-4ea6-bd57-abc25bb8eb1d

Full-parameter fine-tuning can improve the success rate.
8614ce35-e470-48d2-bd24-20810de7746f

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rlinf/models/__init__.py [hunk 1]
```diff
@@ -84,9 +84,25 @@ def get_model(cfg: DictConfig):
...
     return model
 
 
+def _set_module_names(model):
+    """
+    Set the 'name' attribute for all modules in the model.
+    The name is set to the last part of the module's path (e.g., "action_in_proj").
+    This allows FSDP wrap policy to identify modules by name without using id() mapping.
+    """
+    for name, module in model.named_modules():
```
[the remark this answers is not in the document]
Comment (LiuZhihao2022) [reply rlinf/models/__init__.py]: It's not a problem as the root module is always wrapped, so it doesn't need a name for auto_wrap_policy

File: rlinf/hybrid_engines/fsdp/utils.py [hunk 2]
```diff
@@ -489,6 +503,38 @@ def get_grad_norm(
...
+    grad_norm = torch.linalg.vector_norm(
+        torch.stack(
+            [
+                torch.linalg.vector_norm(grad, norm_type, dtype=torch.float32)
+                for grad in grads
+            ],
+        ),
+        norm_type,
+        dtype=torch.float32,
+    )
```
[the remark this answers is not in the document]
Comment (LiuZhihao2022) [reply rlinf/hybrid_engines/fsdp/utils.py:541]: The usage of `torch.linalg.vector_norm` follows the official FSDP implementation and has not been modified.
